### PR TITLE
Feature/prevent xss csrf

### PR DESCRIPTION
**Pull request by solidgoldpig**

### CODE REVIEW

File: lib/server/server.js [hunk 1]
```diff
@@ -125,10 +127,44 @@ const configureMiddleware = async (options = {}) => {
     skip: () => ENV === 'testing'
   }))
 
+  // CORS policy - deny all
```
Comment (solidgoldpig) [inline lib/server/server.js]: CORS and Iframe-busting

File: lib/middleware/routes-static/routes-static.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
 const path = require('path')
 const express = require('express')
 
+const staticOptions = {
```
Comment (solidgoldpig) [inline lib/middleware/routes-static/routes-static.js]: Enable caching of assets

File: lib/middleware/user-session/user-session.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 const express = require('express')
 const router = express.Router()
 const cookieParser = require('cookie-parser')
+// const csrf = require('csurf')
+// const csrfProtection = csrf({cookie: true})
```
Comment (asmega) [inline lib/middleware/user-session/user-session.js]: can delete

File: lib/middleware/user-session/user-session.js [hunk 4]
```diff
@@ -59,6 +61,9 @@ external.init = (options = {}) => {
   }
 
   router.use(cookieParser())
+
+  // router.use(csrfProtection)
```
Comment (asmega) [inline lib/middleware/user-session/user-session.js]: can delete

File: lib/middleware/csrf/csrf.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const init = (token) => {
+  return async (req, res, next) => {
+    if (!req.headers['x-forwarded-host']) {
+      return next()
+    }
+    const userId = req.user.userId
+    const secret = getTokenSecret(userId, token)
+    if (req.method === 'POST') {
+      const contentType = req.headers['content-type']
+      // const contentType = req.get('content-type')
```
Comment (asmega) [inline lib/middleware/csrf/csrf.js]: delete me?

File: lib/middleware/referrer/referrer.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+const {FBError} = require('@ministryofjustice/fb-utils-node')
+const init = (fqd) => {
+  return (req, res, next) => {
+    if (!req.headers['x-forwarded-host'] || !fqd) {
+      return next()
+    }
+    if (req.method === 'POST') {
+      let disallowed
+      let referrer = req.headers['referer']
+      // let referrer = req.get('referrer')
```
Comment (asmega) [inline lib/middleware/referrer/referrer.js]: delete me?

File: lib/middleware/user-session/user-session.js (hunk 3 above)
Comment (solidgoldpig) [reply lib/middleware/user-session/user-session.js]: Done

File: lib/middleware/user-session/user-session.js (hunk 4 above)
Comment (solidgoldpig) [reply lib/middleware/user-session/user-session.js]: Done

File: lib/middleware/csrf/csrf.js (hunk 5 above)
Comment (solidgoldpig) [reply lib/middleware/csrf/csrf.js]: Done

File: lib/middleware/referrer/referrer.js (hunk 6 above)
Comment (solidgoldpig) [reply lib/middleware/referrer/referrer.js]: Done

**Pull request merged**